splinterd/tests/admin/circuit_abandon.rs

```diff
@@ -483,3 +483,336 @@ pub fn test_3_party_circuit_abandon() {
+    // Assert the circuits for the first node are returned as expected
+    assert_eq!(
+        node_a
+            .admin_service_client()
+            .list_circuits(None)
+            .expect("Failed to list circuits")
+            .data
+            .len(),
+        1
+    );
+    assert_eq!(
+        node_a
+            .admin_service_client()
+            .list_circuits(Some("status=abandoned"))
+            .expect("Failed to list circuits")
+            .data
+            .len(),
+        1
+    );
+    // Assert the circuits for the second node are returned as expected
+    assert_eq!(
+        node_b
+            .admin_service_client()
+            .list_circuits(None)
+            .expect("Failed to list circuits")
+            .data
+            .len(),
+        1
+    );
+    assert_eq!(
+        node_b
+            .admin_service_client()
+            .list_circuits(Some("status=abandoned"))
+            .expect("Failed to list circuits")
+            .data
+            .len(),
+        1
+    );
+
+    shutdown!(network).expect("Unable to shutdown network");
+}
+
+/// This test validates the process of committing a circuit between 3 nodes and the process of all
+/// nodes abandoning the committed circuit. The test also validates that Splinter service
+/// transactions succeed on an active, committed circuit and do not succeed once any member has
+/// abandoned the circuit.
+///
+/// 1. Create and commit a circuit between 3 nodes
+/// 2. Create and commit another circuit (that will remain active throughout the test) between the
+///    3 nodes
+/// 3. Stop the second node in the network
+/// 4. Create and submit a `CircuitAbandon` payload to abandon the circuit from one node
+/// 5. Restart the stopped node, the second one in the network
+/// 6. Verify the circuit is still returned as active for the restarted node
+/// 7. Verify the circuit is returned as `Abandoned` by the abandoning node, using `list_circuits`
+///    filtered on the circuit's status (`status=abandoned`, in this case)
+/// 8. Stop the third node in the network
+/// 9. Create and submit a `CircuitAbandon` payload to abandon the circuit from the second node
+/// 10. Restart the stopped node, the third one in the network
+/// 11. Verify the circuit is returned as active for the restarted node
+/// 12. Verify the circuit is returned as `Abandoned` by the abandoning node, using `list_circuits`
+///    filtered on the circuit's status (`status=abandoned`)
+/// 13. Stop the first node in the network
+/// 14. Create and submit a `CircuitAbandon` payload to completely abandon the circuit for all
+///    members, from the third node in the network
+/// 15. Restart the stopped node
+/// 16. Verify the circuit is returned as `Abandoned` for each node, using `list_circuits` filtered
+///     on the circuit's status (`status=abandoned`)
+#[test]
+#[ignore]
+pub fn test_3_party_circuit_abandon_stop() {
+    // Start a 2-node network
```

Comment:
this should say start 3-node network

Reply:
Fixed this. Also added a commit to fix another occurrence of this typo